### PR TITLE
Update to specs actors v0.9.6

### PR DESCRIPTION
**Pull request by arajasek**
- Largely a "follow the pattern" PR
- This introduces the go-state-types
- Updates go-fil-markets to include non-blocking retrieval

### CODE REVIEW

**Review by ZenGround0 — APPROVED** [review]
FYI a new change came in the last day extracting the `adt.Empty` type to go-state-types.  When that merges I can put up another PR on lotus, or push a commit to this one if its still up.

File: api/api_full.go [hunk 1]
```diff
@@ -300,7 +303,7 @@ type FullNode interface {
 	// StateMinerSectors returns info about the given miner's sectors. If the filter bitfield is nil, all sectors are included.
 	// If the filterOut boolean is set to true, any sectors in the filter are excluded.
 	// If false, only those sectors in the filter are included.
-	StateMinerSectors(context.Context, address.Address, *abi.BitField, bool, types.TipSetKey) ([]*ChainSectorInfo, error)
+	StateMinerSectors(context.Context, address.Address, *bitfield.BitField, bool, types.TipSetKey) ([]*ChainSectorInfo, error)
```
[the remark this answers is not in the document]
Comment (arajasek) [reply api/api_full.go:309]: Eeeh, I would've done that if I'd realised it earlier, but I don't think it's worth undoing?

**Review by Kubuxu — APPROVED** [review]

**Review by magik6k — APPROVED** [review]

File: chain/vm/runtime.go [hunk 2]
```diff
@@ -341,7 +341,7 @@ func (rt *Runtime) Send(to address.Address, method abi.MethodNum, m vmr.CBORMars
 	if m != nil {
 		buf := new(bytes.Buffer)
 		if err := m.MarshalCBOR(buf); err != nil {
-			rt.Abortf(exitcode.SysErrInvalidParameters, "failed to marshal input parameters: %s", err)
+			rt.Abortf(exitcode.ErrSerialization, "failed to marshal input parameters: %s", err)
```
Comment (anorth) [inline chain/vm/runtime.go:344]: This is a consensus-affecting change. It should be behind a network version check.

Comment (arajasek) [reply chain/vm/runtime.go:344]: @anorth This one "sneaked" in -- it doesn't break consensus pre-upgrade, (and won't break things moving forward).

Comment (Stebalien) [reply chain/vm/runtime.go:344]: How does it not break consensus? It'll change the error codes (or do those not go on chain?).

**Pull request merged**